File: py/example/vanilladiexample/foo_bar_user.py
from vanilladiexample.foo import Foo
from vanilladiexample.bar import Bar


class FooBarUser:
    def __init__(self, foo: Foo, bar: Bar):
        self.foo = foo
        self.bar = bar

    def dance_with_dependencies(self) -> None:
        self.bar.do_something(self.foo.get_something())

File: cs/example/VanillaDIExample/FooBarUser.cs
namespace VanillaDIExample
{
    public class FooBarUser
    {
        private IFoo Foo { get; }
        private IBar Bar { get; }

        public FooBarUser(IFoo foo, IBar bar) => (Foo, Bar) = (foo, bar);


        public void DanceWithDependencies()
        {
            Bar.DoSomething(Foo.GetSomething());
        }
    }
}